### DISCUSSION TITLE
How to use the trained model to output TopN recommendation results

### DISCUSSION THREAD

**Discussion by PBlithe** [RUCAIBox/RecBole#1619] [Q&A] [answered]
I made a personal dataset and used NeuMF for model training. The training results are as follows:

```
05 Jan 01:25    INFO  best valid : OrderedDict([('recall@10', 0.6547), ('precision@10', 0.0655), ('mrr@10', 0.4819), ('ndcg@10', 0.5236), ('hit@10', 0.6547)])
05 Jan 01:25    INFO  test result: OrderedDict([('recall@10', 0.6511), ('precision@10', 0.0651), ('mrr@10', 0.4805), ('ndcg@10', 0.5217), ('hit@10', 0.6511)])
```

The difference between precision and recall is too far. I think there is a problem with the algorithm, so I want to test whether the precision and recall values are correct with a UserID in the test set.
My question:
1) How to get the data in the test collection, that is, all UserIDs and ItemIDs
2) How to use the trained model to obtain the TopN recommendation list

I use the following code to load the trained model:
```
config, model, dataset, train_data, valid_data, test_data = load_data_and_model(
        model_file='saved/NeuMF-Jan-04-2023_17-54-13.pth',
    )
```
I hope I can get your help. Thank you. (๑*◡*๑)

**Comment by VladHardAndSoftWare** [comment, marked as answer]
You can use code from paragraph 4 (https://www.kaggle.com/code/astrung/recbole-lstm-sequential-for-recomendation-tutorial)

**Reply by PBlithe** [reply]
我在Issue找到了相关的回答，大家可以参考：https://github.com/RUCAIBox/RecBole/issues/1114
